src/pubtools/_marketplacesvm/cloud_providers/base.py

```diff
@@ -146,6 +146,20 @@ def _publish(
                 Defaults to ``False``
         """
 
+    @abstractmethod
+    def _delete(
+        self,
+        push_item: T,
+        **kwargs,
+    ) -> T:
+        """
+        Abstract method for deleting images from a public cloud provider.
+
+        Args:
+            push_item (VMIPushItem)
+                The push item to associate and publish a VM image into a product.
+        """
```

Comment:
If you follow my last suggestion just ensure to name this like `_delete_push_item` or give something a better name, I'll let you decide 😄